### ISSUE TITLE
can't build android version in release mode

### ISSUE THREAD

**Issue by renanyoy** [flutter/flutter#70172]
flutter run --release
```
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Execution failed for task ':app:lintVitalRelease'.                      
> Could not resolve all artifacts for configuration ':app:debugRuntimeClasspath'.
   > Failed to transform libs.jar to match attributes {artifactType=processed-jar, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for JetifyTransform: /Users/renanyoy/git/a2ds/apps/Flutter/leads2keys/build/app/intermediates/flutter/debug/libs.jar.
         > Transform's input file does not exist: /Users/renanyoy/git/a2ds/apps/Flutter/leads2keys/build/app/intermediates/flutter/debug/libs.jar. (See https://issuetracker.google.com/issues/158753935)
```

```
[✓] Flutter (Channel stable, 1.22.3, on Mac OS X 10.15.7 19H2, locale en-FR)
    • Flutter version 1.22.3 at /Users/renanyoy/code/sdk/flutter
    • Framework revision 8874f21e79 (12 days ago), 2020-10-29 14:14:35 -0700
    • Engine revision a1440ca392
    • Dart version 2.10.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/renanyoy/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    • CocoaPods version 1.10.0.rc.1

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 50.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

 
[✓] Connected device (1 available)            
    • Pixel 3a (mobile) • 9BCAY1J68Q • android-arm64 • Android 11 (API 30)
```

the problem is than before seeing the error, I've built an app bundle and pushed it on the google play store
and it seems it's a mixed debug/release version that crash at start, so all our users are stuck with a crashing version
until I get a fix as we can't revert back to an old version on google play console.

thank you if you can speed up the process..

**Comment by renanyoy** [comment]
I think it happens cause of the last gradle update

**Comment by darshankawar** [comment]
@renanyoy 
You are correct. This happens due to AGP 4.0. See [this](https://github.com/flutter/flutter/issues/58247) similar open issue that describes your case.
Closing this as duplicate. If you disagree, write in comments and I'll reopen it.
thanks.

**Comment by renanyoy** [comment]
ok, thank you
fixed with gradle update to 6.7.0 and reverting back build.graddle from 4.0.1 to 3.6.2
```
dependencies {
        classpath 'com.android.tools.build:gradle:3.6.2'
```
